osgp/shared/osgp-ws-secret-management/src/main/resources/schemas/secret-management.xsd

```diff
@@ -1,189 +1,230 @@
-            </xsd:sequence>
-        </xsd:complexType>
-    </xsd:element>
-
-    <xsd:element name="activateSecretsResponse">
-        <xsd:complexType>
-            <xsd:sequence>
-                <xsd:element name="Result" type="tns:OsgpResultType" />
-                <xsd:element name="TechnicalFault" minOccurs="0"/>
-            </xsd:sequence>
-        </xsd:complexType>
-    </xsd:element>
-
-    <xsd:element name="hasNewSecretRequest">
-        <xsd:complexType>
-            <xsd:sequence>
-                <xsd:element name="DeviceId" type="tns:DeviceIdentification" />
-                <xsd:element name="SecretType" type="tns:SecretType" />
-            </xsd:sequence>
-        </xsd:complexType>
-    </xsd:element>
-
-    <xsd:element name="hasNewSecretResponse">
-        <xsd:complexType>
-            <xsd:sequence>
-                <xsd:element name="hasNewSecret" type="xsd:boolean" />
-            </xsd:sequence>
-        </xsd:complexType>
-    </xsd:element>
+            <xsd:element name="SecretTypes" type="tns:SecretTypes"/>
+          </xsd:sequence>
+        </xsd:extension>
+      </xsd:complexContent>
+    </xsd:complexType>
+  </xsd:element>
+
+  <xsd:element name="generateAndStoreSecretsResponse">
+    <xsd:complexType>
+      <xsd:complexContent>
+        <xsd:extension base="tns:abstractResponse">
+          <xsd:sequence>
+            <xsd:element name="TypedSecrets" type="tns:TypedSecrets"/>
+          </xsd:sequence>
+        </xsd:extension>
+      </xsd:complexContent>
+    </xsd:complexType>
+  </xsd:element>
+
+  <xsd:element name="activateSecretsRequest">
+    <xsd:complexType>
+      <xsd:complexContent>
+        <xsd:extension base="tns:abstractRequest">
+          <xsd:sequence>
+            <xsd:element name="SecretTypes" type="tns:SecretTypes"/>
+          </xsd:sequence>
+        </xsd:extension>
+      </xsd:complexContent>
+    </xsd:complexType>
+  </xsd:element>
+
+  <xsd:element name="activateSecretsResponse">
+    <xsd:complexType>
+      <xsd:complexContent>
+        <xsd:extension base="tns:abstractResponse"/>
+      </xsd:complexContent>
+    </xsd:complexType>
+  </xsd:element>
+
+  <xsd:element name="hasNewSecretRequest">
+    <xsd:complexType>
+      <xsd:complexContent>
+        <xsd:extension base="tns:abstractRequest">
+          <xsd:sequence>
+            <xsd:element name="SecretType" type="tns:SecretType"/>
+          </xsd:sequence>
+        </xsd:extension>
+      </xsd:complexContent>
+    </xsd:complexType>
+  </xsd:element>
+
+  <xsd:element name="hasNewSecretResponse">
+    <xsd:complexType>
+      <xsd:complexContent>
+        <xsd:extension base="tns:abstractResponse">
+          <xsd:sequence>
+            <xsd:element name="hasNewSecret" type="xsd:boolean"/>
+          </xsd:sequence>
+        </xsd:extension>
+      </xsd:complexContent>
+    </xsd:complexType>
+  </xsd:element>
 
+  <xsd:element name="correlationUid" type="xsd:string"/>
```

Comment:
In common.xsd in various projects, the correlationUid is defined as:
```xml
  <xsd:simpleType name="CorrelationUid">
    <xsd:restriction base="xsd:normalizedString">
      <xsd:minLength value="20" />
    </xsd:restriction>
  </xsd:simpleType>
```
At first I wondered whether it would be good to use that same definition here. On the other hand, in secret-management it is not created but used as provided, and maybe it is good to keep it as just an `xsd:string` here not to restrict the systems that may be using the secret management service.